crates/ty_server/src/server.rs

```diff
@@ -47,23 +49,40 @@ impl Server {
         in_test: bool,
     ) -> crate::Result<Self> {
         let (id, init_value) = connection.initialize_start()?;
-        let init_params: InitializeParams = serde_json::from_value(init_value)?;
-
-        let AllOptions {
-            global: global_options,
-            workspace: mut workspace_options,
-        } = AllOptions::from_value(
-            init_params
-                .initialization_options
-                .unwrap_or_else(|| serde_json::Value::Object(serde_json::Map::default())),
-        );
 
-        let client_capabilities = init_params.capabilities;
+        let InitializeParams {
+            initialization_options,
+            capabilities: client_capabilities,
+            workspace_folders,
+            ..
+        } = serde_json::from_value(init_value)?;
```

Comment:
Nit: This is a pre-existing issue but let's maybe use `context` to add some extra information what's failing to deserialize
```suggestion
        } = serde_json::from_value(init_value).context("Failed to deserialize the `initializationParams`)?;
```